memory/src/main/scala/filodb.memory/MemFactory.scala

```diff
@@ -212,18 +219,51 @@ class BlockMemFactory(blockStore: BlockManager,
   val fullBlocks = ListBuffer[Block]()
 
   // tracks block currently being populated
-  var currentBlock = blockStore.requestBlock(bucketTime, optionSelf).get
+  var currentBlock = requestBlock()
+
+  private def requestBlock() = blockStore.requestBlock(bucketTime, optionSelf).get
 
   // tracks blocks that should share metadata
   private val metadataSpan: ListBuffer[Block] = ListBuffer[Block]()
 
+  // Last time this factory was used for allocation.
+  private var lastUsedNanos = now
+
+  private def now: Long = System.nanoTime()
```

Comment:
How fast is nanoTime call?  
The other concern is that nanoTime can potentially wrap around, can't it?

Reply:
It's plenty fast. Consider also that the block allocation is relatively infrequent, so not something to be concerned with. The nanotime value can wrap around, which is why modulo comparison (subtraction) is required.